assembly/AbstractInt.ts

```diff
@@ -0,0 +1,23 @@
+import { Codec } from "./Codec";
+
+/** Representation for a Int value in the system. */
+export abstract class AbstractInt<T> implements Codec {
+
+    readonly bitLength: i32;
+    readonly value: T;
+
+    constructor(value: T, bitLength: i32) {
```

Comment:
My idea was to make an AbstractInt more like an abstract class than an abstract interface. The logic in all Integers is the same. If we abstract the utils logic, we will be able to abstract the Integers